### DISCUSSION TITLE
Collabora race condition

### DISCUSSION THREAD

**Discussion by thxrben** [CollaboraOnline/online#10224] [General]
Hey folks,

I am stuck having collabora online on a race condition where the connection is "just closed":

`docker run -t -d -p 127.0.0.1:9980:9980 -e 'aliasgroup1=https://XXX.com:443' --restart always --cap-add MKNOD collabora/code`

I have the 127.0.0.1:9980 proxied outside with https and apache2 as a reverse proxy is also proxying with  https towards the collabora server (websockets too).

When trying to open a document, this happens in the logs of nextcloud:

```
nextcloud-nc-1     | 172.17.0.2 - - [14/Oct/2024:12:50:11 +0000] "GET /index.php/apps/richdocuments/wopi/files/429_ocfbhm7lx6ve?access_token=TOKEN&access_token_ttl=0 HTTP/1.1" 200 2392 "-" "COOLWSD HTTP Agent 24.04.7.2"
nextcloud-nc-1     | 172.17.0.2 - - [14/Oct/2024:12:50:11 +0000] "GET /index.php/apps/richdocuments/wopi/files/429_ocfbhm7lx6ve/contents?access_token=TOKEN&access_token_ttl=0 HTTP/1.1" 200 14419 "-" "COOLWSD HTTP Agent 24.04.7.2"
```
 
 With 172.17.0.2 being the docker IP (on the same host) of collabora.
 
Collabora itself prints this:
```
sh: 1: /usr/bin/coolmount: Operation not permitted
sh: 1: /usr/bin/coolmount: Operation not permitted
sh: 1: /usr/bin/coolmount: Operation not permitted
wsd-00001-00055 2024-10-14 12:50:11.705704 +0000 [ docbroker_002 ] WRN  File system of [/opt/cool/child-roots/1-c5c05716/.] is dangerously low on disk space.| wsd/COOLWSD.cpp:433
frk-00028-00028 2024-10-14 12:50:11.707396 +0000 [ forkit ] WRN  The systemplate directory [/opt/cool/systemplate] is read-only, and at least [/opt/cool/systemplate//etc/hosts] is out-of-date. Will have to copy sysTemplate to jails. To restore optimal performance, make sure the files in [/opt/cool/systemplate/etc] are up-to-date.| common/JailUtil.cpp:585
wsd-00001-00055 2024-10-14 12:50:11.801447 +0000 [ docbroker_002 ] WRN  File system of [/opt/cool/child-roots/1-c5c05716/.] is dangerously low on disk space.| wsd/COOLWSD.cpp:433
wsd-00001-00055 2024-10-14 12:50:12.040789 +0000 [ docbroker_002 ] WRN  ToClient-00c: Unusual race - attempts to transition from SessionState::WAIT_DISCONNECT to SessionState::LIVE| wsd/ClientSession.cpp:144
wsd-00001-00055 2024-10-14 12:50:12.056160 +0000 [ docbroker_002 ] ERR  #34: Read failed, have 0 buffered bytes (ECONNRESET: Connection reset by peer)| net/Socket.hpp:1156
```

I cannot get collabora to respond to nextcloud correctly, and wasnt able to find any logs online that mention "Unusual race" or "SessionState::WAIT_DISCONNECT" at all.

If you have any ideas, please let me know. Thank you!

**Comment by vmiklos** [comment]
If you don't get an answer here, try at https://forum.collaboraonline.com/, which usually has answers.

**Reply by thxrben** [reply]
Yes I will, thank you!

**Comment by thxrben** [comment]
I moved the discussion to the collabora online forum:
https://forum.collaboraonline.com/t/race-condition-connection-is-terminated-nextcloud/3072
